internal/staging/start_testrunner.go

```diff
@@ -32,25 +32,26 @@ func (c *controller) startTesting(queue *s2hv1.Queue) error {
 	}
 
 	// check testing timeout
+	// if timeout, change state to `s2hv1.Collecting`
 	if err := c.checkTestTimeout(queue, testingTimeout); err != nil {
 		return err
 	}
 
+	// check test config
+	// if no test configuration, change state to `s2hv1.Collecting`
 	skipTest, testRunners, err := c.checkTestConfig(queue)
-	if err != nil {
+	if err != nil || skipTest {
 		return err
-	} else if skipTest {
-		return nil
 	}
 
 	// trigger the tests
-	for _, testRunner := range testRunners {
-		if err := c.triggerTest(queue, testRunner); err != nil {
-			return err
+	notTriggeredTest := !queue.Status.IsConditionTrue(s2hv1.QueueTestTriggered)
+	if notTriggeredTest {
+		for _, testRunner := range testRunners {
+			_ = c.triggerTest(queue, testRunner)
```

Comment:
This line fix is what cause an infinite loop if an error occurs.

also added condition not to called runners API if it's doesn't have `pipelineID`

Reply:
marked in this line
https://github.com/agoda-com/samsahai/pull/123/files#diff-be77600073ee9c9ab7b305580a237c21d77ce0e78c68f6f0c6436fab542471e4R86

Reply:
but that marks when fail to get the result?

Reply:
concerns: if there is a network issue, it will not get retried (should split trigger condition by testrunner type)